src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookExplorerViewlet.ts

```diff
@@ -260,40 +259,46 @@ export class NotebookExplorerViewPaneContainer extends ViewPaneContainer {
 		this.validateQuery(query).then(() => {
 			if (this.views.length > 1) {
 				let filesToIncludeFiltered: string = '';
-				this.views.forEach(async (v) => {
+				// search book results first and then notebooks
+				const pinnedNotebookIndex = this.views.findIndex(view => view.id === 'pinnedBooksView');
+				const viewPanes = this.views;
+				viewPanes.push(viewPanes.splice(pinnedNotebookIndex, 1)[0]);
+				viewPanes.forEach(async (v) => {
 					if (v instanceof TreeViewPane) {
 						const { treeView } = (<ITreeViewDescriptor>Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).getView(v.id));
 						if (treeView.dataProvider) {
 							let items = await treeView?.dataProvider.getChildren(treeView?.root);
 							items?.forEach(root => {
 								this.updateViewletsState();
 								if (root.contextValue === 'providedBook' || root.contextValue === 'savedBook') {
-									let rootFolder = URI.file(root.resourceUri.path);
+									const rootFolder = URI.file(root.resourceUri.path);
 									let folderToSearch = { folder: rootFolder };
 									if (root.tooltip.toString().includes('content')) {
 										let pattern = {};
 										pattern['content/**'] = true;
 										folderToSearch['includePattern'] = pattern;
 									}
-									query.folderQueries = query.folderQueries.filter((folder) => {
-										if (!!folder.includePattern && !folder.includePattern.toString().startsWith('content')) {
-											//verify if pinned notebook is not opened in Books Viewlet
-											return path.relative(folder.folder.fsPath, folderToSearch.folder.fsPath) !== '..';
-										} else {
-											return true;
-										}
-									});
 									query.folderQueries.push(folderToSearch);
 									filesToIncludeFiltered = filesToIncludeFiltered + path.join(folderToSearch.folder.fsPath, '**', '*.md') + ',' + path.join(folderToSearch.folder.fsPath, '**', '*.ipynb') + ',';
 								} else {
 									let pattern = {};
-									let rootFolder = URI.file(path.dirname(root.resourceUri.path));
-									let pathToNotebook = isString(root.tooltip) ? root.tooltip : root.tooltip.value;
-									let baseName = path.join('**', path.basename(pathToNotebook));
+									const rootFolder = URI.file(path.dirname(root.resourceUri.path));
+									const baseName = path.join('**', path.basename(root.resourceUri.path)).replace(/\\/g, '/');
+									let isOpenedInBooksView = false;
 									pattern[baseName] = true;
-									let folderToSearch = { folder: rootFolder, includePattern: pattern };
-									query.folderQueries.push(folderToSearch);
-									filesToIncludeFiltered = filesToIncludeFiltered + rootFolder + ',';
+									query.folderQueries.forEach((folder) => {
+										//check for books
```

Comment:
Small nit, usually has space after // and then the comment. 
This and the "//verify..."